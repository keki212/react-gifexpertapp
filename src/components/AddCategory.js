import React, { useState } from 'react'
import PropTypes from "prop-types";

const AddCategory = ({ setCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) =>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        //console.log('handlesubmit', inputValue)
        //setCategory([...categories, inputValue])
        if( inputValue.length > 2){
            setCategory(cats => [ inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
        <p>{ inputValue }</p>
          <input 
              type="text"
              value={ inputValue }
              onChange={handleInputChange}
          /> 
        </form>
    )
}
AddCategory.propTypes ={
    setCategory : PropTypes.func.isRequired
}
export default AddCategory;
