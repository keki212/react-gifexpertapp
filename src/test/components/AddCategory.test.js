import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas del componente <AddCategory.test.js />', () => {    
    
    const setcategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategory={ setcategories }/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategory={ setcategories }/>);
    })

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo.';
        input.simulate('change', { target: { value }});
        expect(wrapper.find('p').text().trim()).toBe(value)
    });
    
    test('No debe enviar la informaciÃ³n con submit', () => {
        wrapper.find('form').simulate('submit',{ preventDefault(){} });

        expect( setcategories ).not.toHaveBeenCalled();
    });

    test('Debe llamar el setCategories y limpiar la caja de texto', () => {
        //const input = wrapper.find('input');
        const value = 'One Pounch';
        wrapper.find('input').simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value);
        wrapper.find('form').simulate('submit',{ preventDefault(){}});
        expect(setcategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
    
    
})
