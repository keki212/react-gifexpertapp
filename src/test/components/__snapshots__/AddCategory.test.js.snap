// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <AddCategory.test.js /> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
